enhancements/machine-config/alignment-with-image-mode-rhel-for-the-mco.md

```diff
@@ -0,0 +1,192 @@
+
+### Goals
+
+* Future Day-2 configuration owned by the MCO will, by default, involve building container images from config specification and layer the image by rpm-ostree rebase / bootc switch. 
+* Enable the user to upgrade their cluster to a rpm-ostree-based image-mode cluster and to a bootc-based image-mode cluster seamlessly 
+
+### Non-Goals
+
+* On Cluster Layering
+* Bootc Day-2 Configuration Tools 
+    - Description: Bootc has opened a door for disk image customization via configmap. Switching from rpm-ostree to bootc, MCO should not only make sure all the functionality remains but also proactively extending its support to make sure all the customization power brought in by bootc are directed to the user, allowing them to fully maximize these advantages.This will involve creating a new user-side API for fetching admin-defined configuration and pairing MCO with a bootc day-2 configuration tool for applying the customizations. Interest and Proposals in which has been raised several times in previous discussions.
+    - Why non-goal: This is a valid request and will enhance user experience by increasing customization option.This is a non-goal for this enhancement because this request should be the next step after bootc switch happening. It will be announced and discussed in detail in a follow-up enhancement. 
+* Splitting the RHCOS Image into layers
+
+## Proposal
+
+The efforts needed to fully align the MCO with Image Mode RHEL in 4.19 or 4.20 can be divided into three crucial phases, each building upon the previous one. In the section below, we will include a detailed description and dependency recognitions for each phase. 
+
+Phase 1 On Cluster Layering GA
+
+On-Cluster Layering introduces a seamlessly image-based configuration flow with details summarized as follows: “perform an on-cluster build (OCB) to combine the base OS image shipped with OpenShift, any admin-provided custom content, and the MachineConfig content managed by the MCO. This image will then be deployed to all of the cluster nodes using rpm-ostree (and later, bootc) to facilitate a singular, transactional, update”
+
+On-Cluster Layering is already shipped in a tech preview version in 4.16. Currently, we are targeting 4.18 for its GA. In this phase, we are aiming that, by the time On Cluster Layering GA, it will stably support most day-2 configurations specified by machine config and ship the change as a container image layer, which include osImageURL specification (DONE), extensions and kernel changes (PLANNED) and file writing to mutable local file system directories (DONE). Another important GA criteria for On Cluster Layering and Phase 1 is that, currently, On Cluster Layering is only a day-2 configuration tool for Openshift, with the workflow being that you install a cluster and then opt in to it. We need to figure out an On Cluster Layering day-1 configuration story so that people are installing a cluster with it.
```

Comment:
```suggestion
On-Cluster Layering is already shipped as a tech preview version in 4.16. Currently, we are targeting 4.18 for its GA. In this phase, we are aiming that, by the time On Cluster Layering GA, it will support most day-2 configurations specified by machine config and ship the change as a container image layer, which include osImageURL specification (DONE), extensions and kernel changes (PLANNED) and file writing to mutable local file system directories (DONE). Another important GA criteria for On Cluster Layering and Phase 1 is that, currently, On Cluster Layering is only a day-2 configuration tool for Openshift, with the workflow being that you install a cluster and then opt in to it. We need to figure out an On Cluster Layering day-1 configuration story so that users can install a cluster with it.
```